requirements.txt

```diff
@@ -1 +1,7 @@
-cocotb
\ No newline at end of file
+numpy
+matplotlib
+cocotb==1.8.0
+cocotb-test==0.2.4
+pytest==7.4.0
+pytest-xdist==3.3.1
```

Comment:
Some of these seem like optional requirements.

To me these ones look nice for everyone to have:
* `cocotb`
* `cocotb-test`
* `pytest`

These ones seem not necessary at this point (add them in a later PR maybe?)?
Why do we need them?
* `remote_pdb`
* `numpy`
* `matplotlib`

This one seems like a nice to have, but not a necessary dependency (we can add them to a `requirements-optional.txt`?
* `pytest-xdist`

Reply:
Hi @JosseVanDelm! The numpy and matplotlib were used by Gui in his setup because he had to plot and process vector data. I originally thought that we might use it later but I guess I'll add them when they are indeed needed. I agree they're not really needed for now. I'll take them off and add the requirements-optional.txt :)